pkg/hive/hive.go

```diff
@@ -260,57 +272,102 @@ func (s *Service) checkAndAddPeers(ctx context.Context, peers pb.Peers) {
 
 	var peersToAdd []swarm.Address
 	mtx := sync.Mutex{}
-	wg := sync.WaitGroup{}
 
-	for _, p := range peers.Peers {
-		err := s.sem.Acquire(ctx, 1)
-		if err != nil {
-			return
-		}
+	if s.bootnode {
+		for _, p := range peers.Peers {
```

Comment:
not sure what is the desirable outcome here. this basically means - if we are a bootnode we persist every peer we get over discovery. if anything, we should not persist _any_ node which we are not connected to

Reply:
because bootnodes do not dial, the newest commit discards the received peers list without storing or validating it.